### ISSUE TITLE
Update Janusgraph to 0.3.0 

### ISSUE THREAD

**Issue by etiennedi** [creativesoftwarefdn/weaviate#744]
We are currently using Janusgraph v0.2.x.

## Why?
We want to (and have to) update to the latest version v.0.3.x because
* our analytics microservice (based in Java) works with the newest versions, but would need more fine tuning to work with older versions as well
* we should always aim to keep our major dependencies up to date

## Risks
We don't see any major obstacles or risks. There could be minor API changes (because the Tinkerpop version increases from 3.2 to 3.3) around the graph management queries. We hope to be able to work them out.

*Update: This turned out to be only half-true. The change in the underlying GraphSON version is a pretty massive breaking change for the API. However, we took a shortcut and downgraded to the old GraphSON version. There is a new issue to clean up this newly created tech debt: #760*

## Todos
* [x] Build new version in creativesoftwarefdn/janusgraph-docker and realease it
  * released as `:0.3.1`
* [x] Incorporate the above into our Docker setup
  * [x] dev setup
  * [x] acceptance test setup
  * [x] runtime stable setup
  * [x] runtime unstable setup 
* [x] Fix anything that doesn't work anymore
  * only minor issues due to GraphSON v1 downgrade
* [x] See if we can / have to update Elasticsearch
  * no
* ~~Update Spark, we already know that Janus 0.3 requires a newer Spark version.~~
  * spark isn't part of our regular setup yet. It will be with #719, so nothing to upgrade yet.

## Further Reading
* [Janusgraph Version Compatibility Matrix](https://docs.janusgraph.org/latest/version-compat.html)

**Comment by etiennedi** [comment]
The PR for this was closed without merging due to the branch restructuring. I'll create a new PR, as this needs to be merged before the eventual PR for #720 .

**Comment by etiennedi** [comment]
Is completed, not sure why we never closed the issue. Closing it now.
